enterprise/models.py

```diff
@@ -541,3 +543,71 @@ def __repr__(self):
         Return uniquely identifying string representation.
         """
         return self.__str__()
+
+
+@python_2_unicode_compatible
+class EnrollmentNotificationEmailTemplate(TimeStampedModel):
+    """
+    Store optional templates to use when emailing users about course enrollment events.
+    """
+
+    class Meta(object):
+        app_label = 'enterprise'
+
+    BODY_HELP_TEXT = _(
+        'Fill in a standard Django template; available variables include `user_name`, '
+        '`course_name`, `program_type`, `course_type`, `organization_name`, `start_date`, '
+        'and `course_url`. All items are strings, except for `start_date`, which is a '
+        '`datetime.datetime` object.'
+    )
+
+    SUBJECT_HELP_TEXT = _(
+        'Fill in a Python format string, using the same variables as are available for '
+        'the HTML and plaintext body template.'
+    )
+
+    plaintext_template = models.TextField(blank=True, help_text=BODY_HELP_TEXT)
+    html_template = models.TextField(blank=True, help_text=BODY_HELP_TEXT)
+    subject_line = models.CharField(max_length=100, blank=True, help_text=SUBJECT_HELP_TEXT)
+    site = models.OneToOneField(Site, related_name="enterprise_enrollment_template")
+    history = HistoricalRecords()
+
+    def render_html_template(self, kwargs):
+        """
+        Render just the HTML template and return it as a string.
+        """
+        return self.render_template(mark_safe(self.html_template), kwargs)
+
+    def render_plaintext_template(self, kwargs):
+        """
+        Render just the plaintext template and return it as a string.
+        """
+        return self.render_template(self.plaintext_template, kwargs)
+
+    def render_templates(self, kwargs):
```

Comment:
If the goal of this function is to render ALL templates, then let's be explicit about that -> `render_all_templates()` -- otherwise I'd expect to be able to send in some sort of switch to control which template(s) are rendered.